system-test/system_test.go

```diff
@@ -15,30 +15,55 @@
 package testing
 
 import (
+	"archive/tar"
 	"bytes"
+	"context"
 	"flag"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"runtime"
 	"strings"
 	"testing"
 	"text/template"
 	"time"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
 	"github.com/google/pprof/profile"
 )
 
 var (
 	binaryHost          = flag.String("binary_host", "", "host from which to download precompiled binaries; if no value is specified, binaries will be built from source.")
 	runOnlyV8CanaryTest = flag.Bool("run_only_v8_canary_test", false, "if true test will be run only with the v8-canary build, otherwise, no tests will be run with v8 canary build")
-	pprofDir            = flag.String("pprof_nodejs_path", "", "path to directory containing pprof-nodejs module")
+	pprofDir            = flag.String("pprof_nodejs_path", "", "the absolute path to directory containing pprof-nodejs module")
+	runOn               = flag.String("run_on", "local", "environment on which to run system test. Either linux-docker, alpine-docker, or local.")
 
 	runID = strings.Replace(time.Now().Format("2006-01-02-15-04-05.000000-0700"), ".", "-", -1)
 )
 
-var tmpl = template.Must(template.New("benchTemplate").Parse(`
+const alpineImage = "node:10-alpine"
+const linuxImage = "node:10"
+
+var dockerTmpl = template.Must(template.New("dockerTemplate").Parse(`
+	FROM {{.Image}}
+
+	{{if .BuildBinary}}
```

Comment:
See my other comments - getting into the business of generating Docker files is an overkill imo, I'd avoid it if possible.

Reply:
Decided that using docker makes most sense when binaries need to be downloaded (it provides a clean slate where the many things installed on kokoro VMs aren't available). and simplified this.